TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/AWSAuthService.java

```diff
@@ -1,43 +1,47 @@
 package org.opensearch.migrations.replay;
 
-import com.amazonaws.secretsmanager.caching.SecretCache;
 import lombok.extern.slf4j.Slf4j;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerAsyncClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
 
 import java.nio.charset.Charset;
 import java.util.Base64;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 @Slf4j
 public class AWSAuthService implements AutoCloseable {
 
-    private final SecretCache secretCache;
+    private final SecretsManagerAsyncClient secretsManagerClient;
 
-    public AWSAuthService(SecretCache secretCache) {
-        this.secretCache = secretCache;
+    public AWSAuthService(SecretsManagerAsyncClient secretsManagerClient) {
+        this.secretsManagerClient = secretsManagerClient;
     }
 
     public AWSAuthService() {
-        this(new SecretCache());
+        this(SecretsManagerAsyncClient.builder().build());
     }
 
     // SecretId here can be either the unique name of the secret or the secret ARN
-    public String getSecret(String secretId) {
-        return secretCache.getSecretString(secretId);
+    public CompletableFuture<GetSecretValueResponse> getSecret(String secretId) {
+        return secretsManagerClient.getSecretValue(builder -> builder.secretId(secretId));
     }
 
     /**
-     * This method returns a Basic Auth header string, with the username:password Base64 encoded
+     * This method synchronously returns a Basic Auth header string, with the username:password Base64 encoded
      * @param username The plaintext username
      * @param secretId The unique name of the secret or the secret ARN from AWS Secrets Manager. Its retrieved value
      *                 will fill the password part of the Basic Auth header
      * @return Basic Auth header string
      */
-    public String getBasicAuthHeaderFromSecret(String username, String secretId) {
-        String authHeaderString = username + ":" + getSecret(secretId);
+    public String getBasicAuthHeaderFromSecret(String username, String secretId) throws ExecutionException, InterruptedException {
+        String secretValue = getSecret(secretId).get().secretString();
```

Comment:
I don't like having the synchronous method alongside the async one and then being opinionated on exceptions that can come out of the synchronous method that would only happen when dealing with something asynchronous.

You could also return a CompletableFuture by applying the getSecret Output through a finalizer that does everything else.  That let's the caller determine how to handle the exceptions and keeps these methods more consistent.